### ISSUE TITLE
Properly handle "MySQL server has gone away"

### ISSUE THREAD

**Issue by conrad10781** [Sannis/node-mysql-libmysqlclient#179]
I have a process that is intended to run for a long time. It's often processing several requests a second, but in some cases it can be > 1 day in between a request, which is resulting in a timeout of the connection. 

Originally, when the longest time between requests was less than a day, I changed the MySQL wait_timeout value to 86400. I want to avoid making this number any bigger and am looking for a better way to handle this error.

I apologize if this is a duplicate of #163 , but I honestly couldn't understand the response in that issue, or at least how to modify it to resolve my problem.

In summary, is it possible to provide a snippet on how to deal with this issue ( once the problem happens ) or a way maybe to check before doing any queries if we have a valid connection ( before the problem )?
